examples/quickstart/file_serialization.cpp

```diff
@@ -0,0 +1,139 @@
+//  Copyright (c) 2017 Hartmut Kaiser
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+// This example demonstrates how the HPX serialization archives could be used
+// to directly store/load to/from a file.
+
+#include <hpx/hpx.hpp>
+#include <hpx/hpx_main.hpp>
+#include <hpx/traits/serialization_access_data.hpp>
+
+#include <cstddef>
+#include <cstring>
+#include <iostream>
+#include <fstream>
+#include <stdexcept>
+#include <string>
+#include <vector>
+
+struct file_wrapper
+{
+    file_wrapper(std::string const& name, std::ios_base::openmode mode)
+      : stream_(name.c_str(), mode), mode_(mode)
+    {
+        if (!stream_.is_open())
+            throw std::runtime_error("Couldn't open file: " + name);
+
+        std::fstream::pos_type fsize = stream_.tellg();
+        stream_.seekg(0, std::ios::end);
+        data_.resize(stream_.tellg() - fsize);
+        stream_.seekg(fsize, std::ios::beg);
+
+        if (mode & std::ios_base::in)
+        {
+            stream_.read(data_.data(), data_.size());
+        }
+    }
+
+    ~file_wrapper()
+    {
+        if (mode_ & std::ios_base::out)
+        {
+            stream_.write(data_.data(), data_.size());
+        }
+    }
+
+    void read(void* address, std::size_t count, std::size_t current) const
+    {
+        std::memcpy(address, &data_[current], count);
+    }
+
+    void write(void const* address, std::size_t count, std::size_t current)
+    {
+        std::memcpy(&data_[current], address, count);
+    }
+
+    std::size_t size() const
+    {
+        return data_.size();
+    }
+
+    void resize(std::size_t count)
+    {
+        data_.resize(data_.size() + count);
+    }
+
+private:
+    std::vector<char> data_;
+    std::fstream stream_;
+    std::ios_base::openmode mode_;
+};
+
+namespace hpx { namespace traits
+{
+    template <>
+    struct serialization_access_data<file_wrapper>
+      : default_serialization_access_data<file_wrapper>
+    {
+        static std::size_t size(file_wrapper const& cont)
+        {
+            return cont.size();
```

Comment:
Could this function return cont.size_data() if "size_data()" was defined in file_wrapper?